src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -371,15 +451,25 @@ private boolean canReadFrom(Job<?, ?> job, AbstractBuild<?, ?> build) {
         return b;
     }
 
-    // retrieve the "folder" (jenkins root if no folder used) for this build
-    private ItemGroup getItemGroup(AbstractBuild<?, ?> build) {
-        ItemGroup group = build.getProject().getRootProject().getParent();
-        return group;
+    private Job<?, ?> getRootJob(Job<?, ?> job) {
+        if (job instanceof AbstractProject) {
+            return ((AbstractProject<?,?>)job).getRootProject();
+        } else {
+            return job;
+        }
+    }
 
+    // retrieve the "folder" (jenkins root if no folder used) for this build
+    private ItemGroup getItemGroup(Run<?, ?> build) {
+        if (build instanceof AbstractBuild) {
+            return ((AbstractBuild) build).getProject().getRootProject().getParent();
+        } else {
+            return build.getParent().getParent();
```

Comment:
Can be simplified with the method you just added:

    return getRootProject(build.getParent()).getParent();